pkg/cloud/store/openstack/store.go

```diff
@@ -38,6 +39,21 @@ func (s *Store) Get(key string) (model.OpenStack, bool) {
 	defer s.mtx.RUnlock()
 
 	value, ok := s.clouds[key]
+	if !ok {
+		// Try to get Cloud provider info from Etcd
+		// Check the issue: https://github.com/vCloud-DFTBA/faythe/issues/23
+		r, err := s.etcdcli.DoGet(common.Path(model.DefaultCloudPrefix, key))
+		if err != nil || len(r.Kvs) != 1 {
```

Comment:
what is the case that `len(r.Kvs) > 1`? should it be `len(r.Kvs) == 0`

Reply:
A cloud should be unique, if there are more than one cloud, something went wrong here. Just return false.